app/models/seance.rb

```diff
@@ -1,7 +1,21 @@
 class Seance < ApplicationRecord
   belongs_to :hall
   belongs_to :movie
+  has_many :reservations
 
   validates :movie_id, presence: true
   validates :hall_id, presence: true
+
+  def available_seats
```

Comment:
`available_seats` and `taken_seats` probably should be in repository

Reply:
Since I used these methods in representer isn't it clear to have it in the model? Or should I pass the repository to the representer as well?

Reply:
This is a reason for the repository
have all queries related do context in one place